Java testing

Thunae opened the pull request:
The framework for Java Tests for all the servlets. 
This changes how the servlets operate and uses Mockito and JUnit to test the functionality and features of the servlets.

Thunae committed:
Framework for Mockito and JUnit Testing

Clean up tests after Rebasing from master

Add more tests for datastore and login

Address Catherine PR Comments: remove initmocks and change imports

Move static imports and change to camelCase

Debugging to bring to speed with master

Framework for Mockito and JUnit Testing

Clean up tests after Rebasing from master

Add more tests for datastore and login

Debugging to bring to speed with master

Rearrange Imports, Change test names to camelCase

Thunae committed:
Framework for Mockito and JUnit Testing

Thunae committed:
Framework for Mockito and JUnit Testing

Clean up tests after Rebasing from master

Add more tests for datastore and login

Address Catherine PR Comments: remove initmocks and change imports

Move static imports and change to camelCase

Debugging to bring to speed with master

Thunae committed:
Clean up tests after Rebasing from master

Thunae committed:
Add more tests for datastore and login

Thunae committed:
Address Catherine PR Comments: remove initmocks and change imports

Thunae committed:
Move static imports and change to camelCase

Thunae committed:
Debugging to bring to speed with master

Thunae committed:
Minor Fixes to make master function correctly (#40)

* Minor Fixes to make master function correctly

* Spacing

* Reimplement arrow function, change jquery selector

Thunae committed:
Add Map to the resultsPage (#36)

* Add Map to the resultsPage

* Formatting updates, arrow function

Thunae committed:
Rebase to master

lcatherine commented on src/test/java/com/google/sps/ConvertLocationServletTest.java:
```diff
@@ -0,0 +1,125 @@
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import java.lang.String;
```
this is out of order

Thunae replied:
done

lcatherine commented on src/test/java/com/google/sps/LoginServletTest.java:
```diff
@@ -0,0 +1,102 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.ArgumentMatchers.*;
+import static org.junit.Assert.assertEquals;
```
static block comes first, also no * allowed

Thunae replied:
done

lcatherine commented on src/test/java/com/google/sps/QueryServletTest.java:
```diff
@@ -0,0 +1,239 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.any;
+import static org.junit.Assert.assertEquals;
```
static blocks come first

Thunae replied:
done

lcatherine commented:
In general: please camelCase all of your test cases and static blocks come first in the imports

lcatherine commented on src/main/java/com/google/sps/servlets/QueryServlet.java:
```diff
@@ -46,16 +48,27 @@
-            servletResponse.getWriter().println(gson.toJson(new Response("NO_RESULTS", null)));
-        else if (response.getStatus().equals("OK"))
+        if(response == null){
+            response = new Response("NO_RESULTS", new ArrayList<Restaurant>());
```
make sure changing this to an empty list doesn't impact anything calling this method. null != an empty arraylist

Thunae replied:
A null response will produce an error in the front end when we try to index it. This is meant to change the status to NO_RESULTS because the JS has an condition to handle an invalid response with no results in it

Thunae committed:
Rearrange Imports, Change test names to camelCase

Thunae committed:
Advanced algorithm (#37)

* Add weighted algorithm chooseRestaurant to queryservlet

* Fix variable names

* Align function names

* Make naming consistent, change Assert.assert

* Addition of the Past Searches page, updates to the SearchServlet, and addition of the FeedbackServlet. This fetches the user's past searches, formats them on the html page, and adds a modal through which the user can submit feedback on the restaurant.  (#33)

* fix merge conflicts, add changes

* Past searches cards are added

* restaurant name carries through to feedback form

* feedback servlet form submission completed, servlet is fully functional

* UI updates + remove commented out lines, add extra line at end of file

* feedback servlet changes imported

* start adding comments, fix merge conflicts

* imported changes from updatedSearches branch, add comments

* fix order of import statement, extra comment signs, and unnecessary comments

* add spaces, fix function syntax, delete duplicated file

Co-authored-by: ryan <ryangudal@gmail.com>

* Secrets Manager Implementation (#35)

* Merge

* Merge

* Merge

* Restaurant Images routed through servlet to protect API key

* Secret Manager Class, Route Geolocation and Images through servlets

* Address PR comments: make access secret a singleton

* IOException caught

* Style header in caps

* Make AccessSecret member strings final

* Remove debug statements, add imports

* Address PR comments, add spacing, rework expressions

* Remove debug statements from secrets

* spacing fixes

* addition of account page updates to the correct branch (cherrypickBranch) of past searches updates  (#34)

* elements added to account page

* imported changes from updatedSearches branch, add comments

* UI update

* removed target and old (unused) files from git

* fix issue from merge, change syntax of async function

* fix function syntax

* Minor Fixes to make master function correctly (#40)

* Minor Fixes to make master function correctly

* Spacing

* Reimplement arrow function, change jquery selector

* Add Map to the resultsPage (#36)

* Add Map to the resultsPage

* Formatting updates, arrow function

* Add weighted algorithm chooseRestaurant to queryservlet

Fix variable names

Align function names

Make naming consistent, change Assert.assert

spacing fixes

Add URL encoder, fix spacing

* Bug fix: add user so call is defined

Co-authored-by: evd123 <46075902+evd123@users.noreply.github.com>

Thunae committed:
Clean after Rebasing from Master

Thunae committed:
Rebase Master and Java Testing
